cranelift/codegen/src/isa/x64/mod.rs

```diff
@@ -230,156 +230,8 @@ fn isa_constructor(
 #[cfg(test)]
 mod test {
     use super::*;
-    use crate::cursor::{Cursor, FuncCursor};
-    use crate::ir::{types::*, RelSourceLoc, SourceLoc, UserFuncName, ValueLabel, ValueLabelStart};
-    use crate::ir::{AbiParam, Function, InstBuilder, JumpTableData, Signature};
-    use crate::isa::CallConv;
     use crate::settings;
     use crate::settings::Configurable;
-    use core::str::FromStr;
-    use cranelift_entity::EntityRef;
-    use target_lexicon::Triple;
-
-    /// We have to test cold blocks by observing final machine code,
-    /// rather than VCode, because the VCode orders blocks in lowering
-    /// order, not emission order. (The exact difference between the
-    /// two is that cold blocks are sunk in the latter.) We might as
-    /// well do the test here, where we have a backend to use.
-    #[test]
-    fn test_cold_blocks() {
```

Comment:
Could we add a generic cold-blocks filetest? It looks like we have a few related to bugs (`atomic-cas-bug.clif`, `br_table.clif`, and a few legalizer/verifier tests too) but none that try to generally exercise the feature like this test does.